Content.Server/Traits/Assorted/ParacusiaSystem.cs

```diff
@@ -0,0 +1,65 @@
+using Content.Shared.Audio;
+using Content.Server.Camera;
+using Robust.Shared.Random;
+using Robust.Shared.Player;
+using Robust.Shared.Map;
+
+namespace Content.Server.Traits.Assorted;
+
+public sealed class ParacusiaSystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly CameraRecoilSystem _camera = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ParacusiaComponent, ComponentStartup>(SetupParacusia);
+    }
+
+    private void SetupParacusia(EntityUid uid, ParacusiaComponent component, ComponentStartup args)
+    {
+        component.NextIncidentTime =
+            _random.NextFloat(component.TimeBetweenIncidents.X, component.TimeBetweenIncidents.Y);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        foreach (var paracusia in EntityQuery<ParacusiaComponent>())
+        {
+            paracusia.NextIncidentTime -= frameTime;
+
+            if (paracusia.NextIncidentTime >= 0)
+                continue;
+
+            // Set the new time.
+            paracusia.NextIncidentTime +=
+                _random.NextFloat(paracusia.TimeBetweenIncidents.X, paracusia.TimeBetweenIncidents.Y);
```

Comment:
instead of a decrement like this called on update, you should set NextIncidentTime to the current round time + the time between incidents, and then just check if the current round time >= NextIncidentTime